spec/controllers/transactions_invoice_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::Transactions::InvoicesController do
+  describe "GET associated invoice" do
+    it "retrieves the correct invoice for that transaction" do
+      transaction = FactoryGirl.create(:transaction,      
+        id: 1,
+        invoice: FactoryGirl.create(:invoice, id: 20)
+      )
+      10.times do
+        FactoryGirl.create(:invoice)
+      end
+      
+      get :show, params: { format: :json, id: 1 }
```

Comment:
@concach 
See my comment in spec/controllers/merchants_invoices_controller_spec.rb.

Also on line 28 of this file.